app/src/main/java/com/keylesspalace/tusky/ViewMediaActivity.kt

```diff
@@ -153,7 +154,7 @@ class ViewMediaActivity : BaseActivity(), ViewImageFragment.PhotoActionsListener
         window.statusBarColor = Color.BLACK
         window.sharedElementEnterTransition.addListener(object : NoopTransitionListener {
             override fun onTransitionEnd(transition: Transition) {
-                (adapter as SharedElementTransitionListener).onTransitionEnd()
+                adapter.onTransitionEnd(viewPager.currentItem)
```

Comment:
I'm not 100% sure that it will be current item but should be fine for common cases.